### PR TITLE
Rename `yii\base\InvalidParamException` to `yii\base\InvalidArgumentException`

### LINKED ISSUE #13080: Rename InvalidParamException to InvalidArgumentException?
Seems like there is misuse of parameter term. :)

See "Parameters and arguments" section of https://en.wikipedia.org/wiki/Parameter_(computer_programming)

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | no
| Breaks BC?    | yes
| Tests pass?   | yes
| Fixed issues  | #13080

First of all, I'm not completely sure if `InvalidArgumentException` is the best name, maybe better to use `InvalidArgException` for brevity (previous name was shortened - `InvalidParamException`, not `InvalidParameterException`). But, from the other side, many dependent libraries use `InvalidArgumentException` (including `Doctrine` for example). Just try autocomplete it in IDE.

Additional fixes:

- Reordered changes for 2.1 in CHANGELOG.
- Refactored long lines (more than 120 chars) related to this change.
- Removed unused import of `yii\base\InvalidParamException` in one place.

TODO:

- [ ] Add this change to yii2-faker. (https://github.com/yiisoft/yii2-faker/blob/master/FixtureController.php#L219) Requires creation of another branch.
- [ ] Check other extensions for using `yii\base\InvalidParamException`.

Closes #13080

### CODE REVIEW

File: framework/CHANGELOG.md [hunk 1]
```diff
@@ -9,12 +9,13 @@ Yii Framework 2 Change Log
 - Enh #12938: Allow to pass additional parameters to `yii\base\View::renderDynamic()` (mikehaertl)
 - Enh #13006: Added a `/` to the `yii\captcha\Captcha::$captchaAction` string to work correctly in a module also (boehsermoe)
 - Removed methods marked as deprected in 2.0.x (samdark)
+- Chg #7770: Updated the fallback date formats used when no `intl` extension is installed to match the defaults of the latest ICU version (cebe)
```
Comment (SilverFire) [inline framework/CHANGELOG.md]: File is broken after merge

**Review by SilverFire — CHANGES_REQUESTED** [review]
I'm fine with InvalidArgumentException.

**Comment by SilverFire** [comment]
I suggest leave `yii\base\InvalidParamException` class, but extend it from `yii\base\InvalidArgumentException` for compatibility reasons

File: framework/CHANGELOG.md (hunk 1 above)
Comment (samdark) [inline framework/CHANGELOG.md]: Yes, diff is definitely wrong...

**Comment by samdark** [comment]
It is 2.1.0, we can break things.

**Comment by SilverFire** [comment]
I know that we can, but it's a massive BC break that will affect a lot of code.
Making too many hard-to-upgrade changes can spurn developers from upgrade at all

**Comment by rob006** [comment]
> I suggest leave yii\base\InvalidParamException class, but extend it from yii\base\InvalidArgumentException for compatibility reasons

We can do this in 2.0.x and remove `InvalidParamException` in 2.1.0.

**Comment by SilverFire** [comment]
Probably

**Comment by arogachev** [comment]
> I suggest leave `yii\base\InvalidParamException` class, but extend it from `yii\base\InvalidArgumentException` for compatibility reasons

Interesting idea, but I think it will cause excess confusion.

> I know that we can, but it's a massive BC break that will affect a lot of code.
Making too many hard-to-upgrade changes can spurn developers from upgrade at all

If it's covered in `UPGRADE.md` I think it's fine. As a developer you need to understand all consequences when migrating to newer major version.

File: framework/CHANGELOG.md (hunk 1 above)
Comment (arogachev) [inline framework/CHANGELOG.md]: No, it's OK.

> Additional fixes:
> - Reordered changes for 2.1 in CHANGELOG.

I reordered changes for 2.1 to preserve order in dedicated sections like it was done in 2.0.11. I thinj we need to add a note about modifying a changelog according to that and be more accurate when reviewing PRs.

Comment (arogachev) [inline framework/CHANGELOG.md]: Should I revert the changes related with reordering?

Comment (samdark) [inline framework/CHANGELOG.md]: Ideally yes. Changelog is automatically reordered before release so there's no need to do so manually cause it produces weird diffs.

**Review by samdark — APPROVED** [review]

**Review by SilverFire — APPROVED** [review]

**Review by dynasource — APPROVED** [review]

**Comment by cebe** [comment]
> Check other extensions for using yii\base\InvalidParamException.

```
$ grep -rni --exclude=vendor/ InvalidParamException extensions/
extensions/httpclient/Client.php:12:use yii\base\InvalidParamException;
extensions/httpclient/Client.php:121:     * @throws InvalidParamException on invalid format name.
extensions/httpclient/Client.php:140:                throw new InvalidParamException("Unrecognized format '{$format}'");
extensions/httpclient/Client.php:156:     * @throws InvalidParamException on invalid format name.
extensions/httpclient/Client.php:169:                throw new InvalidParamException("Unrecognized format '{$format}'");
extensions/jui/DatePicker.php:11:use yii\base\InvalidParamException;
extensions/jui/DatePicker.php:180:            } catch(InvalidParamException $e) {
extensions/mongodb/rbac/MongoDbManager.php:12:use yii\base\InvalidParamException;
extensions/mongodb/rbac/MongoDbManager.php:592:            throw new InvalidParamException("Cannot add '{$parent->name}' as a child of itself.");
extensions/mongodb/rbac/MongoDbManager.php:596:            throw new InvalidParamException('Cannot add a role as a child of a permission.');
extensions/mongodb/QueryBuilder.php:14:use yii\base\InvalidParamException;
extensions/mongodb/QueryBuilder.php:147:                throw new InvalidParamException('"key" is required for index specification');
extensions/mongodb/QueryBuilder.php:589:     * @throws InvalidParamException if the condition is in bad format
extensions/mongodb/QueryBuilder.php:606:            throw new InvalidParamException('Condition should be an array.');
extensions/mongodb/QueryBuilder.php:665:     * @throws InvalidParamException if wrong number of operands have been given.
extensions/mongodb/QueryBuilder.php:670:            throw new InvalidParamException("Operator '$operator' requires two operands.");
extensions/mongodb/QueryBuilder.php:728:     * @throws InvalidParamException if wrong number of operands have been given.
extensions/mongodb/QueryBuilder.php:733:            throw new InvalidParamException("Operator '$operator' requires three operands.");
extensions/mongodb/QueryBuilder.php:760:     * @throws InvalidParamException if wrong number of operands have been given.
extensions/mongodb/QueryBuilder.php:765:            throw new InvalidParamException("Operator '$operator' requires two operands.");
extensions/mongodb/QueryBuilder.php:840:     * @throws InvalidParamException if wrong number of operands have been given.
extensions/mongodb/QueryBuilder.php:845:            throw new InvalidParamException("Operator '$operator' requires two operands.");
extensions/mongodb/QueryBuilder.php:865:     * @throws InvalidParamException if wrong number of operands have been given.
extensions/mongodb/QueryBuilder.php:870:            throw new InvalidParamException("Operator '$operator' requires two operands.");
extensions/mongodb/QueryBuilder.php:887:     * @throws InvalidParamException if wrong number of operands have been given.
extensions/mongodb/QueryBuilder.php:892:            throw new InvalidParamException("Operator '$operator' requires two operands.");
extensions/mongodb/QueryBuilder.php:911:                throw new InvalidParamException("Unsupported operator '{$operator}'");
extensions/mongodb/file/Upload.php:13:use yii\base\InvalidParamException;
extensions/mongodb/file/Upload.php:173:            throw new InvalidParamException("Unable to read file '{$filename}'");
extensions/sphinx/MatchBuilder.php:10:use yii\base\InvalidParamException;
extensions/sphinx/MatchBuilder.php:176:            throw new InvalidParamException("Operator '$operator' requires three or more operands.");
extensions/sphinx/MatchBuilder.php:200:            throw new InvalidParamException("Operator '$operator' requires exactly one operand.");
extensions/sphinx/MatchBuilder.php:218:            throw new InvalidParamException("Operator '$operator' requires three operands.");
extensions/sphinx/MatchBuilder.php:236:            throw new InvalidParamException("Operator '$operator' requires two operands.");
extensions/sphinx/MatchBuilder.php:250:     * @throws InvalidParamException on invalid operands count.
extensions/sphinx/MatchBuilder.php:255:            throw new InvalidParamException("Operator '$operator' requires two operands.");
extensions/redis/ActiveQuery.php:11:use yii\base\InvalidParamException;
extensions/redis/ActiveQuery.php:373:     * @throws \yii\base\InvalidParamException
extensions/redis/ActiveQuery.php:492:        throw new InvalidParamException('Unknown fetch type: ' . $type);
extensions/redis/LuaScriptBuilder.php:10:use yii\base\InvalidParamException;
extensions/redis/LuaScriptBuilder.php:294:            throw new InvalidParamException("Operator '$operator' requires exactly one operand.");
```

**Comment by cebe** [comment]
The faker and imagine extensions have a yiisoft/yii2 requirement of `*` that needs to be fixed.

**Comment by cebe** [comment]
Fixed in https://github.com/yiisoft/yii2-faker/commit/8a406289c3af4da5e0092a96b37734dc1a16a43b for faker. The `*` requirement only affects the 2.0.x branch of imagine extension, not the 2.1 one.

File: framework/CHANGELOG.md (hunk 1 above)
Comment (cebe) [inline framework/CHANGELOG.md]: @samdark actually if devs would put their issues into the correct order right away we would not have the weird reordering issue on merge :)

Comment (cebe) [inline framework/CHANGELOG.md]: reordering of issue inside of a PR will cause problems though.

**Review by cebe — APPROVED** [review]

File: framework/CHANGELOG.md (hunk 1 above)
Comment (arogachev) [inline framework/CHANGELOG.md]: Thanks for the explanation, I will keep this in mind in the future.

**Comment by arogachev** [comment]
Reverted CHANGELOG, is it ready to merge?

**Comment by cebe** [comment]
Merged, thank you!

**Pull request merged**